### PR TITLE
add dispense translations to a cancellation response

### PR DESCRIPTION
**Pull request by conner-jowitt**

### CODE REVIEW

File: coordinator/src/services/translation/cancellation/cancellation-practitioner-role.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export function createPractitionerRole(
     practitioner: createReference(practitionerReference),
     organization: createReference(organizationReference),
     code: createJobRoleNameCode(hl7AgentPerson.code._attributes.code),
-    telecom: toArray(hl7AgentPerson.telecom)[0]._attributes ? convertTelecom(hl7AgentPerson.telecom) : undefined
+    telecom: toArray(hl7AgentPerson.telecom)[0]?._attributes ? convertTelecom(hl7AgentPerson.telecom) : undefined
```
Comment (mfjarvis) [inline coordinator/src/services/translation/cancellation/cancellation-practitioner-role.ts:19]: I don't think that question mark is going to do anything. You'll still get an index out of bounds error if the array is empty

File: coordinator/src/services/translation/cancellation/cancellation-response.ts [hunk 2]
```diff
@@ -85,7 +85,41 @@ function createBundleEntries(cancellationResponse: CancellationResponse) {
...
+      fhirPractitionerRole: fhirPerformerPractitionerRole,
+      fhirOrganization: fhirPerformerOrganization
+    } = convertAgentPerson(performerAgentPerson)
+    const performerPractitionerId = fhirPerformerPractitioner.id
+    const performerOrganizationCode = performerAgentPerson.representedOrganization.id._attributes.extension
+    const performerOrganizationName = performerAgentPerson.representedOrganization.name._text
+    fhirMedicationRequest.dispenseRequest = createDispenserInfoReference(
+      performerPractitionerId, performerOrganizationCode, performerOrganizationName
+    )
+    bundleResources.push(fhirPerformerPractitioner, fhirPerformerPractitionerRole, fhirPerformerOrganization)
```
Comment (mfjarvis) [inline coordinator/src/services/translation/cancellation/cancellation-response.ts:103]: As far as I can tell, we only need the Practitioner in the bundle. The others appear to be unreferenced

Comment (mfjarvis) [reply coordinator/src/services/translation/cancellation/cancellation-response.ts:103]: Actually looking at the IG, `MedicationReference.dispenseRequest.performer.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-DispensingPerformer").valueReference` should point to the PractitionerRole, not the Practitioner (and therefore the Practitioner and Organization would all be referenced too)

File: coordinator/src/services/translation/cancellation/cancellation-practitioner-role.ts (hunk 1 above)
Comment (mfjarvis) [reply coordinator/src/services/translation/cancellation/cancellation-practitioner-role.ts:19]: Well looks like I'm wrong

Comment (mfjarvis) [reply coordinator/src/services/translation/cancellation/cancellation-practitioner-role.ts:19]: JS is weird yo

**Review by mfjarvis — APPROVED** [review]

**Review by ri-ch — APPROVED** [review]

**Pull request merged**